app/script/conversation/ConversationRepository.js

```diff
@@ -568,17 +582,28 @@ z.conversation.ConversationRepository = class ConversationRepository {
   /**
    * Get conversation with a user.
    * @param {User} user_et - User entity for whom to get the conversation
+   * @param {string} [team_id] - Team ID in which the conversation should be searched
    * @returns {Promise} Resolves with the conversation with requested user
    */
-  get_one_to_one_conversation(user_et) {
+  get_one_to_one_conversation(user_et, team_id = this.active_team().id) {
     for (const conversation_et of this.conversations()) {
-      if ([z.conversation.ConversationType.ONE2ONE, z.conversation.ConversationType.CONNECT].includes(conversation_et.type())) {
+      if (team_id) {
+        if (conversation_et.type() === z.conversation.ConversationType.REGULAR) {
+          if (user_et.id === conversation_et.participating_user_ids()[0] && team_id === conversation_et.team_id) {
+            return Promise.resolve(conversation_et);
+          }
+        }
+      } else if ([z.conversation.ConversationType.ONE2ONE, z.conversation.ConversationType.CONNECT].includes(conversation_et.type())) {
         if (user_et.id === conversation_et.participating_user_ids()[0]) {
```

Comment:
`user_et.id === conversation_et.participating_user_ids()[0]` appears twice so it actually makes more sense..!